src/main/java/codeu/model/data/User.java

```diff
@@ -14,61 +14,109 @@
-     */
-    public User(UUID id, String name, String passwordHash, Instant creation) {
-        this.id = id;
-        this.name = name;
-        this.passwordHash = passwordHash;
-        this.creation = creation;
-    }
-
-    /**
-     * Returns the ID of this User.
-     */
-    public UUID getId() {
-        return id;
-    }
-
-    /**
-     * Returns the username of this User.
-     */
-    public String getName() {
-        return name;
-    }
-
-    /**
-     * Returns the password hash of this User.
-     */
-    public String getPasswordHash() {
-        return passwordHash;
-    }
-
-    /**
-     * Returns the creation time of this User.
-     */
-    public Instant getCreationTime() {
-        return creation;
-    }
 
+  private final UUID id;
+  private final String name;
+  private final String passwordHash;
+  private final Instant creation;
+  private Set<String> friends;
+  private UserStore userStore;
+
+  /**
+   * Constructs a new User.
+   *
+   * @param id the ID of this User
+   * @param name the username of this User
+   * @param passwordHash the password hash of this User
+   * @param creation the creation time of this User
+   * @param friends a set with the usernames of the users friends
+   */
+  public User(UUID id, String name, String passwordHash, Instant creation, Set<String> friends) {
+    this.id = id;
+    this.name = name;
+    this.passwordHash = passwordHash;
+    this.creation = creation;
+    this.friends = friends;
+    userStore = UserStore.getInstance();
+  }
+
+  /**
+   * Constructs a new User.
+   *
+   * @param id the ID of this User
+   * @param name the username of this User
+   * @param passwordHash the password hash of this User
+   * @param creation the creation time of this User
+   */
+  public User(UUID id, String name, String passwordHash, Instant creation) {
+    this(id, name, passwordHash, creation, new HashSet<>());
+  }
+
+  /** Returns the ID of this User. */
+  public UUID getId() {
+    return id;
+  }
+
+  /** Returns the username of this User. */
+  public String getName() {
+    return name;
+  }
+  
+  /** Returns the password hash of this User. */
+  public String getPasswordHash() {
+    return passwordHash;
+  }
+
+  /** Returns the creation time of this User. */
+  public Instant getCreationTime() {
+    return creation;
+  }
+
+  /** Returns a set with the usernames of this User's friends. */
+  public Set<String> getFriends() {
+    return friends;
+  }
+
+  /**
+   * Adds friend with given id if User with given id exists
+   *
+   * @param username the username of the friend to be added
+   */
+  public void addFriend(String username) {
+      if (userStore.getUser(id) != null) {
+        friends.add(username);
+      }
+  }
+
+  /**
+   * Removes friend with given id if exists
+   *
+   * @param username the username of the friend to be added
+   */
+  public void removeFriend(String username) {
+      friends.remove(username);
+  }
+
```

Comment:
Your comment says removes friend with given id if exists, but there isn't a check so it might error if you don't have some kind of if statement to make sure it exists.

Reply:
I initially used id as the parameter instead of the username, I'll change the comments and commit together with my other changes. I first need to merge this so I can finally pull from master and finish my changes to open a new pr. I'll fix those small things on this next pr.